pallets/xcmp-queue/src/lib.rs

```diff
@@ -405,7 +411,10 @@ impl<T: Config> Pallet<T> {
 			XcmpMessageFormat::ConcatenatedEncodedBlob => {
 				while !remaining_fragments.is_empty() {
 					last_remaining_fragments = remaining_fragments;
-					if let Ok(blob) = <Vec<u8>>::decode(&mut remaining_fragments) {
+					if let Ok(blob) = <Vec<u8>>::decode_all_with_depth_limit(
+						32, // hardcoded constant since we don't process any blob XCMP msgs anyway
```

Comment:
How did you choose this number? :D

Reply:
Completely random, it's simply based on powers of 2.

Reply:
Then why not just `decode_all`?